cmake/extensions.cmake

```diff
@@ -1711,56 +1711,201 @@ endfunction()
 #                          Issue an error for any relative path not specified
 #                          by user with `-D<path>`
 #
+# CONF_FILES <path>: Find all configuration files in path and return them in a
+#                    list. Configuration files will be:
+#                    - DTS:       Overlay files (.overlay)
+#                    - Kconfig:   Config fragments (.conf)
+#                    The conf file search will return existing configuration
+#                    files for the current board.
+#                    CONF_FILES takes the following additional arguments:
+#                    DTS <list>:   List to populate with DTS overlay files
+#                    KCONF <list>: List to populate with Kconfig fragment files
+#                    BUILD <type>: Build type to include for search.
+#                                  For example:
+#                                  BUILD debug, will look for <board>_debug.conf
+#                                  and <board>_debug.overlay, instead of <board>.conf
+#
 # returns an updated list of absolute paths
 function(zephyr_file)
-  set(options APPLICATION_ROOT)
-  cmake_parse_arguments(FILE "${options}" "" "" ${ARGN})
-  if(NOT FILE_APPLICATION_ROOT)
+  set(file_options APPLICATION_ROOT CONF_FILES)
+  if((ARGC EQUAL 0) OR (NOT (ARGV0 IN_LIST file_options)))
     message(FATAL_ERROR "No <mode> given to `zephyr_file(<mode> <args>...)` function,\n \
-Please provide one of following: APPLICATION_ROOT")
+Please provide one of following: APPLICATION_ROOT, CONF_FILES")
+  endif()
+
+  if(${ARGV0} STREQUAL APPLICATION_ROOT)
+    set(single_args APPLICATION_ROOT)
+  elseif(${ARGV0} STREQUAL CONF_FILES)
+    set(single_args CONF_FILES DTS KCONF BUILD)
+  endif()
+
+  cmake_parse_arguments(FILE "${options}" "${single_args}" "" ${ARGN})
```

Comment:
This doesn't look right; don't you mean `"${file_options}"` instead of `"${options}"`? How was this tested?

I don't see how `options` is set in this function when this line is reached. Am I missing something?

Reply:
Because `${options}` should be deleted.
So in this case it's actually a good thing it would be empty when it's not defined.

`${options}` removed